### PR TITLE
Add Draw Cone.

### PR DESCRIPTION
**Pull request by jiannanya**
1.Add draw cone.
  2.Add draw cone filled.
  3.Add Imgui example for drawing cone.

### CODE REVIEW

File: im3d.cpp [hunk 1]
```diff
@@ -619,6 +619,86 @@ void Im3d::DrawArrow(const Vec3& _start, const Vec3& _end, float _headLength, fl
 	ctx.end();
 }
 
+void Im3d::DrawCone(const Vec3& _origin, const Vec3& _normal,float height, float _radius, int _detail){
+
+    Context& ctx = GetContext();
+    #if IM3D_CULL_PRIMITIVES
+        if (!ctx.isVisible(_origin, _radius))
```
Comment (john-chapman) [inline im3d.cpp:626]: I think this should be the bounding sphere of the cone: `_origin + _normal * height / 2`, `height / 2`?

**Review by john-chapman — CHANGES_REQUESTED** [review]
Looks great! Just need to use the proper cone bounds for culling and detail estimation.

**Comment by jiannanya** [comment]
Yeah,  I have fixed it for culling and detail estimation.

**Comment by jiannanya** [comment]
> Looks great! Just need to use the proper cone bounds for culling and detail estimation.
Yeah, I have fixed it for culling and detail estimation.

**Comment by john-chapman** [comment]
Great stuff - thanks for your contribution!

**Comment by john-chapman** [comment]
FYI @jiannanya I've revisited the implementation of `DrawCone()` and `DrawConeFilled()` and made some changes to make the API more consistent with the other high order shape functions. I've also added separate start/end radii parameters for drawing truncated cones, and bool args for enabling the start/end caps when drawing the filled variants.

The old versions are still the default behavior to avoid API breaking changes, but are now deprecated. This is currently controlled via `IM3D_USE_DEPRECATED_DRAW_CONE` in [im3d_config.h](https://github.com/john-chapman/im3d/blob/master/im3d_config.h). 

See [the commit here](https://github.com/john-chapman/im3d/commit/3fd7afaf3192ed50bca29c191969aa41252b53d5).

**Pull request merged**